src/main.ts

```diff
@@ -0,0 +1,124 @@
+import os = require('os');
+
+import * as core from '@actions/core';
+import * as github from '@actions/github';
+import * as io from '@actions/io';
+import * as toolCache from '@actions/tool-cache';
+import * as rustCore from '@actions-rs/core';
+
+function getPlatformMatchingTarget(): string {
+    const platform = os.platform() as string;
+    switch (platform) {
+        case "linux":
+            return "x86_64-unknown-linux-gnu"
+        case "win32":
+            return "x86_64-pc-windows-msvc"
+        case "darwin":
+            return "x86_64-apple-darwin"
+        default:
+            throw new Error("Unsupported runner");
+    }
+}
+
+function optionIfValueProvided(option: string, value?: string): string {
+    return value ? ` ${option} ${value}` : '';
+}
+
+function getCheckReleaseArguments(): string[] {
+    return [
+        optionIfValueProvided('--package', rustCore.input.getInput('crate-name')),
+        optionIfValueProvided('--manifest-path', rustCore.input.getInput('manifest-path')),
+        rustCore.input.getInputBool('verbose') ? ' --verbose' : ''
+    ].filter(el => el != '');
+}
+
+function getGitHubToken(): string {
+    const token = process.env['GITHUB_TOKEN'] || rustCore.input.getInput('github-token');
+    if (!token)
+        throw new Error('Querying the GitHub API is possible only if the GitHub token is set.');
+    return token;
+}
+
+async function getCargoSemverChecksDownloadURL(target: string): Promise<string> {
+    const octokit = github.getOctokit(getGitHubToken());
+
+    const getReleaseUrl = await octokit.rest.repos.getLatestRelease({
+        owner: 'obi1kenobi',
+        repo: 'cargo-semver-checks'
+    });
+
+    const asset = getReleaseUrl.data.assets.find(asset => {
+        return asset['name'].endsWith(`${target}.tar.gz`)
```

Comment:
Is it possible the release isn't always a `.tar.gz`? Windows in particular seems to be a `.zip`.

Reply:
There is also a release for Windows packed as `.tar.gz` and it seems to be exactly the same binary as the one in the `.zip` file, am I right? If yes, then it is easier to always download the `.tar.gz` archive as it doesn't require separating the cases.

Reply:
Good call, I didn't notice that release.